api/restHandler/PipelineConfigRestHandler.go

```diff
@@ -1028,10 +1031,34 @@ func (handler PipelineConfigRestHandlerImpl) GetDeploymentTemplate(w http.Respon
 				writeJsonResp(w, err, nil, http.StatusInternalServerError)
 				return
 			}
+
 			if pg.ErrNoRows == err {
-				template.ChartRefId = chartRefId
-				template.Id = 0
-				template.Latest = false
+				appOverride, err := handler.chartService.GetAppOverrideForDefaultTemplate(chartRefId)
+				if err != nil {
+					handler.Logger.Errorw("service err, GetDeploymentTemplate", "err", err, "appId", appId, "chartRefId", chartRefId)
+					writeJsonResp(w, err, nil, http.StatusInternalServerError)
+					return
+				}
+				mapB, _ := json.Marshal(appOverride)
+				if err != nil {
+					handler.Logger.Errorw("marshal err, GetDeploymentTemplate", "err", err, "appId", appId, "chartRefId", chartRefId)
+					return
+				}
+				bytes, err := json.Marshal(template.DefaultAppOverride)
+				if err != nil {
+					handler.Logger.Errorw("marshal err, GetDeploymentTemplate", "err", err, "appId", appId, "chartRefId", chartRefId)
+					return
+				}
+				withCombinedPatch, err := jsonpatch.MergePatch(mapB, bytes)
+				if err != nil {
+					panic(err)
```

Comment:
in case of panic will this result in system failure? and no response is written here...can we handle this error and write response accordingly?